doc/ref_arch/openstack/chapters/chapter04.md

```diff
@@ -363,10 +363,31 @@ Swift is the object storage management service, Swift depends on Keystone and po
+| neutron metadata agent (neutron-metadata-agent) | The metadata service provides a way for instances to retrieve instance-specific data. The networking service, neutron, is responsible for intercepting these requests and adding HTTP headers which uniquely identify the source of the request before forwarding it to the metadata API server. These functions are performed by the neutron metadata agent. | Optional | Network node </br> (Controller node if no network node present) |
+| neutron plugin agent (neutron-\*-agent) | Runs on each compute node to control and manage the local virtual network driver (such as the Open vSwitch or Linux Bridge) configuration and local networking configuration for VMs hosted on that node. | Required | Every Compute Node |
+<p align="center"><b>Table 4-2: Neutron Services Placemen</b></p>
+
+**Issues with the standard networking (centralized routing) approach**
+
+The network node performs both routing and NAT functions and represents both a scaling bottleneck and a single point of failure. 
+
+Two VMs on different compute nodes and using different project networks where the both of the project networks are connected by a project router. For communication between the two VMs (instances with a fixed or floating IP address), the network node routes east-west network traffic among project networks using the same project router. Even though the instances are connected by a router, all routed traffic must flow through the network node, and this becomes a bottleneck for the whole network.
+
+While the separation of the routing function from the controller node to the network node provides a degree of scaling it is not a truly scalable solution.  We can add additional cores/compute-power to the network node, but, eventually, it runs out of processing power. Therefore, for scaled deployments, use of Dynamic Virtual Routing (DVR) is advised. With DVR, each compute node also hosts the L3-agent (providing the distributed router capability) and this then allows direct instance to instance (East-West) communications. 
+
+**Distributed Virtual Routing (DVR)**
+
+The OpenStack “[High Availability Using Distributed Virtual Routing (DVR)]( https://docs.openstack.org/liberty/networking-guide/scenario-dvr-ovs.html)” provides an in depth view into how DVR works and the traffic flow between the various nodes and interfaces for three different use cases. Please note that DVR was introduced in the OpenStack Juno release and, thus, its detailed analysis in the Liberty release documentation is not out of character for OpenStack documentation. 
+
+DVR addresses both scalability and high availability for some L3 functions but is not fully fault tolerant. For example, North/South SNAT traffic is vulnerable to single node (network node) failures. [DVR with VRRP]( https://docs.openstack.org/neutron/pike/admin/config-dvr-ha-snat.html) addresses this vulnerability and is, thus, the preferred OpenStack Networking solution. 
+ 
```

Comment:
@pgoyal01 , Many thanks for your great effort in this section,

A general comment, we need to emphasis on the usage of SDN to solve the above problem and we need to have a section/title cover the SDN integration with Neutron API ( either ML2/ML3 or Monoltic ), which we can develop it with you here 
@sukhdevkapur

Reply:
@ASawwaf It makes sense to include SDN in the networking discussion.
@pgoyal01 - I provided this feedback via email. We need to include a sub-section on SDN Networking. I agree with @ASawwaf that SDN does solve most of the networking issues. If you need the content (a paragraph or so( on SDN to be included here, I will be happy to provide. Please let me know.